crates/revive-types/src/primitives.rs

```diff
@@ -0,0 +1,226 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), allow(unused_imports))]
+
+use alloc::vec::Vec;
+use frame_support::weights::Weight;
+use ink_primitives::H160;
+use pallet_revive_uapi::ReturnFlags;
+use scale::{
+    Decode,
+    Encode,
+    MaxEncodedLen,
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+    DispatchError,
+    RuntimeDebug,
+    traits::{
+        Saturating,
+        Zero,
+    },
```

Comment:
`sp_runtime` introduces too many dependencies for essentially just getting `DispatchError`, so may be better to: 
- Get `RuntimeDebug` from `sp_runtime_debug` directly
- `Saturating` and `Zero` from `sp_arithmetic` directly
- Copy `DispatchError` minimally [like subxt](https://github.com/paritytech/subxt/blob/b13b8ec8d0e9b86ea5f475452fd1ea9396922108/subxt/src/error/dispatch_error.rs#L16-L66)

Reply:
Not sure how worth is remove sp_runtime here, we have to bring sp_runtime_debug, sp_arithmetic and have more copied code `DispatchError`

Reply:
Gotcha! We can revisit in future PRs if it makes sense.